api/v1beta1/nodenetworkconfigurationenactment_types.go

```diff
@@ -33,12 +33,13 @@ type NodeNetworkConfigurationEnactment struct {
 	Status shared.NodeNetworkConfigurationEnactmentStatus `json:"status,omitempty"`
 }
 
-func NewEnactment(nodeName string, policy NodeNetworkConfigurationPolicy) NodeNetworkConfigurationEnactment {
+func NewEnactment(node *corev1.Node, policy NodeNetworkConfigurationPolicy) NodeNetworkConfigurationEnactment {
 	enactment := NodeNetworkConfigurationEnactment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: shared.EnactmentKey(nodeName, policy.Name).Name,
+			Name: shared.EnactmentKey(node.Name, policy.Name).Name,
 			OwnerReferences: []metav1.OwnerReference{
-				{Name: policy.Name, Kind: policy.TypeMeta.Kind, APIVersion: policy.TypeMeta.APIVersion, UID: policy.UID},
+				{Name: policy.Name, Kind: policy.Kind, APIVersion: policy.APIVersion, UID: policy.UID},
```

Comment:
Does not having two reference mean that it will be removed only when both of these are gone? While what we want to do is to remove it as soon as one of them is gone. Please keep me honest here.

Reply:
It should based on a little google research of mine. 
[E2E](https://github.com/nmstate/kubernetes-nmstate/blob/main/test/e2e/handler/nncp_cleanup_test.go) tests should keep us covered as we're testing NNCEs are cleaned up on NNCP deletion

Reply:
Oh, that's awesome then, nice cleanup. Sorry about the raised hold.

Reply:
No need to apologize, looks like your concerns are correct!
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/808/pull-kubernetes-nmstate-e2e-handler-k8s/1429712164502900736

Reply:
Bless the tests \o/

Reply:
yep, you're right!